docs/source/api_ref_models.rst

```diff
@@ -2,105 +2,409 @@
 torchtune.models
 ================
 
-.. currentmodule:: torchtune.models
 
-llama3
+This page contains information on all model architectures available in the core torchtune library.
+
+Llama3
 ------
 
-All models from the `Llama3 family <https://llama.meta.com/llama3/>`_.
+.. currentmodule:: torchtune.models.llama3
+
+`Llama 3 <https://llama.meta.com/llama3/>`_ - released in April of 2024 - is an auto-regressive language model that uses an optimized transformer architecture.
+The tuned versions use supervised fine-tuning (SFT) and reinforcement learning with human feedback (RLHF) to align with human preferences for helpfulness and safety.
+The model was pretrained on over **15 trillion tokens** of data from publicly available sources. The pretraining data has a cutoff of March 2023 for the 8B and December 2023
+for the 70B models respectively. The fine-tuning data includes publicly available instruction datasets, as well as over **10M human-annotated examples**.
+You can read more about the model from the official `model card <https://github.com/meta-llama/llama3/blob/main/MODEL_CARD.md#model-details>`_.
+
+Download the model
+^^^^^^^^^^^^^^^^^^
+
+Pre-trained models can be downloaded from the Hugging Face Hub with the following command:
 
 .. code-block:: bash
 
-    tune download meta-llama/Meta-Llama-3-8B-Instruct --hf-token <ACCESS_TOKEN>
+    tune download meta-llama/Meta-Llama-3-8B-Instruct --output-dir /tmp/Meta-Llama-3-8B-Instruct --hf-token <ACCESS_TOKEN>
+
+You can find all model checkpoints on the `official Meta Llama model page <https://huggingface.co/collections/meta-llama/meta-llama-3-66214712577ca38149ebb2b6>`_.
+
+Model architectures
+^^^^^^^^^^^^^^^^^^^
+
+These are models utilizing the default sizes released by the Meta team.
+
+.. autosummary::
+    :toctree: generated/
+    :nosignatures:
+
+    llama3_8b
+    lora_llama3_8b
+    qlora_llama3_8b
+    llama3_70b
+    lora_llama3_70b
+
+If you want to experiment with different sizes of models or train a model from scratch, we also provide generic builder
+functions for our models. These builders **will not** work out-of-the-box with pretrained checkpoints.
 
+.. autosummary::
+    :toctree: generated/
+    :nosignatures:
+
+    llama3
+    lora_llama3
+
+Tokenizer
+^^^^^^^^^
 
 .. autosummary::
     :toctree: generated/
     :nosignatures:
 
-    llama3.llama3_8b
-    llama3.llama3_70b
-    llama3.lora_llama3_8b
-    llama3.qlora_llama3_8b
-    llama3.lora_llama3_70b
+    llama3_tokenizer
```

Comment:
worth linking to something like https://github.com/meta-llama/llama3/blob/main/llama/tokenizer.py for a one-file implementation?